### ISSUE TITLE
How to optimize?

### ISSUE THREAD

**Issue by lxq0204** [kevinlin311tw/caffe-cvprw15#37]
I have read the paper''Deep Learning of Binary Hash Codes for Fast Image Retrieval'', but I have some questions about optimization problem . I want to apply this method to the classification problem using tensorflow framwork.
1、What is the activation function of the F8 layer?  Softmax ?
2、What is the loss function? 
3、How to optimize ?
